### PR TITLE
Update copyright years to 2024.

### PR DESCRIPTION
**Pull request by 0xSalikh**
----
📚 Documentation preview 📚: https://cpython-previews--113608.org.readthedocs.build/

### CODE REVIEW

File: Include/internal/mimalloc/mimalloc.h [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* ----------------------------------------------------------------------------
-Copyright (c) 2018-2023, Microsoft Research, Daan Leijen
+Copyright (c) 2018-2024, Microsoft Research, Daan Leijen
```
Comment (benjaminp) [inline Include/internal/mimalloc/mimalloc.h]: These minmalloc files should not be touched.

**Review by benjaminp — DISMISSED** [review]
Please only touch files similar to previous iterations of this change (like 11f99323c2ae0ec428c370a335695e3d8d4afc1d). We shouldn't be modifying the license headers of files from third party sources.

File: Include/internal/mimalloc/mimalloc.h (hunk 1 above)
Comment (0xSalikh) [reply Include/internal/mimalloc/mimalloc.h]: Corrected. Thank you for your comment)

**Comment by ned-deily** [comment]
One way or another, we should ensure that the copyright years are updated before the next release, which will likely be a 3.13 alpha in a few days.

**Comment by hugovk** [comment]
I've updated this PR to remove the year changes to third-party copyrights, and to add 2024 to the PSF's copyright in `LICENSE`.

CLA: @0xSalikh please could you click the button at https://github.com/python/cpython/pull/113608#issuecomment-1873086967 to sign the CLA?

Even if they don't sign the CLA, I think this is trivial enough to merge without it (but it would be easier if it was).

**Review by ned-deily — APPROVED** [review]
LGTM

**Comment by hugovk** [comment]
Updated to use 2001-2024 year range in LICENSE as suggested in https://github.com/python/cpython/pull/101706: https://github.com/python/cpython/pull/113608/commits/9d2f2ddb49d3060ca33f0cf01acb0b2829aa7bce

**Review by ned-deily — APPROVED** [review]

**Comment by gpshead** [comment]
IMNSHO We should stop doing this.

Trusted Non-PSF legal advice I've received in the past has been that nobody should bother to change dates in a Copyright (C) statement.

**Unless the PSF's current legal council specifically advises us to do this, lets stop updating these dates in the future.** (If they agree that we should not, I'd also ask them if we could get rid of the confusing "-XXXX" through style date at the same time to avoid people wrongly thinking they're out of date and re-proposing the same change in the future)

**Comment by pombredanne** [comment]
> IMNSHO We should stop doing this.
> 
> Trusted Non-PSF legal advice I've received in the past has been that nobody should bother to change dates in a Copyright (C) statement.
> 
> **Unless the PSF's current legal council specifically advises us to do this, lets stop updating these dates in the future.** (If they agree that we should not, I'd also ask them if we could get rid of the confusing "-XXXX" through style date at the same time to avoid people wrongly thinking they're out of date and re-proposing the same change in the future)

@gpshead I agree ++. Updating these dates is mostly harmless, and eventually useless busywork with neither legal nor value. I am talking from experience, as I have some software legal and license compliance chops.

**Pull request merged**